pinot-query-runtime/src/test/resources/queries/QueryHints.json

```diff
@@ -125,6 +125,14 @@
         "description": "Colocated JOIN with partition column and group by non-partitioned column with stage parallelism",
         "sql": "SET stageParallelism=2; SELECT {tbl1}.name, SUM({tbl2}.num) FROM {tbl1} /*+ tableOptions(partition_function='hashcode', partition_key='num', partition_size='4') */ JOIN {tbl2} /*+ tableOptions(partition_function='hashcode', partition_key='num', partition_size='4') */ ON {tbl1}.num = {tbl2}.num GROUP BY {tbl1}.name"
       },
+      {
+        "description": "Broadcast JOIN without partition hint",
+        "sql": "SELECT /*+ joinOptions(left_exchange_type = 'local', right_exchange_type = 'broadcast') */ {tbl1}.num, {tbl1}.name, {tbl2}.num, {tbl2}.val FROM {tbl1} JOIN {tbl2} ON {tbl1}.num = {tbl2}.num"
```

Comment:
Shouldn't be better to create a new hint for exchanges instead of assigning it to the join?

The way I'm suggesting the query would be something like:
```sql
SELECT 
    {tbl1}.num, {tbl1}.name, {tbl2}.num, {tbl2}.val
FROM {tbl1} /*+ exchangeOption(type = 'local') */
JOIN {tbl2} /*+ exchangeOption(type = 'broadcast') */
ON {tbl1}.num = {tbl2}.num;
```

This could also be used for example in aggregates. For example, we could write something like:
```sql

SELECT
    {tbl1}.num, count(*)
from {tbl1} /*+ exchangeOption(type = 'local') */
GROUP BY {tbl1}.num;
```

Reply:
Good suggestion. Let me try it and see if I can make it work

Reply:
Hmm, I'm not able to make it work because this is not really a `TABLE_SCAN` option, or an option applied to any specific `RelNode`. The left and right side of a `JOIN` could be any `RelNode`, and it could be another chained `JOIN`. Do you see a way to extract this hint?

Reply:
Are we not using this?
https://calcite.apache.org/javadocAggregate/org/apache/calcite/rel/hint/package-summary.html

Calcite has good support for hint extraction and propagation.

Reply:
@siddharthteotia We are following this, but the actual problem is that: left and right exchange should be associated with JOIN instead of table. One example would be: there is no guarantee that the JOIN is applied to 2 tables. It could also be applied to the result of a query, or another JOIN etc.

Reply:
And something like:
```sql
   select /*+ NO_HASH_JOIN, RESOURCE(mem='128mb', parallelism='24') */
   from
     (select * from emp limit 10) as emp /*+ INDEX(idx1, idx2) */
     join
     dept /*+ PROPERTIES(k1='v1', k2='v2') */
     on emp.deptno=dept.deptno
```

doesn't work?

Reply:
It can be associated to the `TableScan` node, but the abstraction is not clear to me. We are deciding the exchange distribution for `Join`, but not `TableScan`, and also the child of `Join` might not be table scan